chroot-bin/volume

```diff
@@ -0,0 +1,178 @@
+    #       ^                     ^                 ^ substitute letters
+    #       ^                     ^ substitude spaces for nothing
+    #       ^ select this line only
+}
+
+get_current_output_infos() {
+    # This function gets line from cras_test_client --dump_server_info that 
+    # represents the current state of the current output.
+
+    # The current output informations will not duplicate an output node, this makes
+    # it safe(-ish) to filter out on the current output ID.
+    # This would also select the "Selected Output Node" line, but by filtering the
+    # lines beginning with space characters, this selects the proper line.
+    # If for any reason this starts failing, it could be replaced with a sed script
+    # that acts only between the two "headers" "Output Nodes" and "Input Devices".
+    cras_test_client --dump_server_info | grep "^[[:space:]].*$(get_current_output)"
+}
+
+get_current_volume() {
+    get_current_output_infos | sed -e \
+        "s#[[:space:]]\+#;#g" | cut -d';' -f3
+    #    ^                        ^ Use cut to get the third field (Volume)
+    #    ^ normalize spaces/tab separators to semicolons
+}
+
+set_volume() {
+    local volume="$1"
+    # Clamp volume to zero
+    # A negative amount of absolute volume sets it to 100%
+    if [ "$volume" -lt 0 ] ; then
+        volume=0
+    fi
+    cras_test_client --set_node_volume "$(get_current_output):$volume"
+}
+
+get_is_muted() {
+    local muted="$(host-dbus dbus-send --print-reply --system \
+        --dest=org.chromium.cras --type=method_call \
+        /org/chromium/cras org.chromium.cras.Control.GetVolumeState \
+        | sed -n -e '6s/.*boolean[[:space:]]\+//p')"
+    [ "$muted" = "true" ]
+}
+
+toggle_mute() {
+    if get_is_muted; then
+        unmute
+    else
+        mute
+    fi
+}
+
+mute() {
+    cras_test_client --user_mute 1
+}
+unmute() {
+    cras_test_client --user_mute 0
+}
+
+relative_volume() {
+    local delta="$1"
+    if get_is_muted; then
+        unmute
+        if [ "$delta" -lt 0 ]; then
+            set_volume 0
+        fi
+    else
+        set_volume "$(( $(get_current_volume) + delta ))"
+    fi
+}
+
+print_help() {
+    local script="$(basename "$0")"
+    echo "$script [set] [0-100]"
+    echo "$script up|down [0-100]"
+    echo "$script mute [set|unset|toggle]"
+    echo ""
+    echo "This script changes the volume of the current output"
+    echo "device, as would changing it with the shortcut keys or"
+    echo "with the GUI in Chrome OS."
+    echo ""
+    echo "Shortcut invocations:"
+    echo "  $script up     increases by $DEFAULT_VOLUME_DELTA"
+    echo "  $script down   decreases by $DEFAULT_VOLUME_DELTA"
+    echo "  $script mute   sets muted"
+    echo "  $script 0-100  sets volume to given value"
+}
+
+if [ $# -lt 1 ]; then
+    print_help
+    exit 0
+fi
+
+cmd="$1"
+shift
+
+if [ "$cmd" = "help" ] || [ "$cmd" = "-h" ] || [ "$cmd" = "--help" ]; then
```

Comment:
You may want to turn whole block into a case statement (against ${cmd#-}), so you can do nice things like `h*)` as a sloppy condition for help...